keras/src/models/sequential.py

```diff
@@ -226,12 +228,16 @@ def call(self, inputs, training=None, mask=None):
             # `outputs` are the outputs of `layer` applied to `inputs`. At the
             # end of each iteration `inputs` is set to `outputs` to prepare for
             # the next layer.
-            kwargs = {}
+            layer_kwargs = {}
             if layer._call_has_mask_arg:
-                kwargs["mask"] = mask
+                layer_kwargs["mask"] = mask
             if layer._call_has_training_arg and training is not None:
```

Comment:
Can this be removed in favor of the loop below? You might need to remove the explicit `"training"` arg above so that it comes through `kwargs` instead.

Reply:
Tried doing this, however, this breaks a couple of tests.

Upon investigating, it turns out some places in the code pass the `training` argument as a positional argument to the `call` method of the `Sequential`. This means instead of being caught in the `kwargs`, it gets bound to `mask`.

We should probably avoid doing this as a part of this PR. This can probably be taken up as future cleanup.

Reply:
Sounds good.